packages/build/src/plugins/compatibility.js

```diff
@@ -51,7 +51,23 @@ const siteDependenciesTest = function (allowedSiteDependencies, { packageJson: {
 
 const siteDependencyTest = function ({ dependencyName, allowedVersion, siteDependencies }) {
   const siteDependency = siteDependencies[dependencyName]
-  return typeof siteDependency === 'string' && satisfies(siteDependency, allowedVersion)
+  if (typeof siteDependency !== 'string') {
+    return false
+  }
+
+  // if this is a valid version we can apply the rule directly
+  if (validVersion(siteDependency)) {
+    return satisfies(siteDependency, allowedVersion)
+  }
+
+  try {
+    // if this is a range we need to get the exact version
+    // eslint-disable-next-line node/global-require,import/no-dynamic-require
+    const { version } = require(`${dependencyName}/package.json`)
```

Comment:
https://github.com/netlify/build/blob/master/packages/build/src/utils/resolve.js should be used instead, so that the `cwd` is correct.

At the moment, this look up this file from wherever `@netlify/build` is installed. Instead, it should look it up from the site's build directory, i.e. `buildDir`.

Reply:
Please let me know if https://github.com/netlify/build/pull/2408/commits/2f2c7fd3558226d5d2d8ab157e6b72045f03358d is what you had in mind

Reply:
Yes, this looks good!

I have just added a fix to use `p-locate` (async `find()`) instead of `p-filter` (async `filter()`) since there might be several `compatibility` conditions, and we want to return the first matching one. 5534d59735ac2ee06d7f2aa26fa395121e340801